src/main/java/com/tms/lesson44/hw/dao/StudentDao.java

```diff
@@ -0,0 +1,37 @@
+package com.tms.lesson44.hw.dao;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.tms.lesson44.hw.model.Student;
+import com.tms.lesson44.hw.repository.StudentsRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StudentDao {
```

Comment:
DAO and Repository - are 2 names for the same level layer of the application, so it's not correct to inject one of them inside another.
Rename to service